diff --git a/dtls_config.h b/dtls_config.h
new file mode 100644
index 0000000..b1109d3
--- /dev/null
+++ b/dtls_config.h
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ *
+ * Copyright (c) 2011, 2012, 2013, 2014, 2015 Olaf Bergmann (TZI) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompanies this distribution.
+ *
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at 
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * Contributors:
+ *    Olaf Bergmann  - initial API and implementation
+ *    Hauke Mehrtens - memory optimization, ECC integration
+ *
+ *******************************************************************************/
+
+/**
+ * @file dtls_config.h
+ * @brief public tinydtls API
+ */
+
+#ifndef _DTLS_CONFIG_H_
+#define _DTLS_CONFIG_H_
+
+#ifndef HAVE_SYS_TIME_H
+#define HAVE_SYS_TIME_H
+#endif
+
+#ifndef HAVE_TIME_H
+#define HAVE_TIME_H
+#endif
+
+#ifndef HAVE_ASSERT_H
+#define HAVE_ASSERT_H
+#endif
+
+#ifndef WITH_SHA256
+#define WITH_SHA256
+#endif
+
+#endif /* _DTLS_CONFIG_H_ */


